src/mococrw/ecies.h

```diff
@@ -0,0 +1,400 @@
+     * @param keySize The key size in bytes
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setSymmetricCipherKeySize(size_t keySize);
+
+    /**
+     * @brief Sets a function for creating the AES object inside ECIES
+     *
+     * Default: If no function is set the default value is AES-CBC PKCS7-padding IV=0x00
+     *
+     * As the key for AES is generated during ECIES key generation, the AES object has to be instantiated after that.
+     * This function is a dependency injection.
+     * (e.g. for AES-CBC PKCS7-padding IV=0x00:
+     \code{.cpp}
+        auto cipherDecFunc = [zeroIV](const std::vector<uint8_t> &key) -> std::unique_ptr<AESCipher> {
+            return AESCipherBuilder(SymmetricCipherMode::CBC, SymmetricCipherKeySize::S_256, key)
+                    .setIV(std::vector<uint8_t>(AESCipherBuilder::getDefaultIVLength(mode)))
+                    .setPadding(SymmetricCipherPadding::PKCS)
+                    .buildDecryptor();
+        };
+
+        auto cipherEncFunc = [zeroIV](const std::vector<uint8_t> &key) -> std::unique_ptr<AESCipher> {
+            return AESCipherBuilder(SymmetricCipherMode::CBC, SymmetricCipherKeySize::S_256, key)
+                    .setIV(std::vector<uint8_t>(AESCipherBuilder::getDefaultIVLength(mode)))
+                    .setPadding(SymmetricCipherPadding::PKCS)
+                    .buildEncryptor();
+        };
+    \endcode
+     * @param func A std::function which returns an object implementing the SymmetricCipherI interface
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setSymmetricCipherFactoryFunction(std::function<std::unique_ptr<SymmetricCipherI>(
+                                                           const std::vector<uint8_t>&)> func);
+
+
+    /**
+     * @brief Sets the salt for the key derivation function (KDF).
+     *
+     * If no value is set, the KDF calculation is done without any salt
+     * @param kdfSalt The salt value used for key derivation.
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setKDFSalt(std::vector<uint8_t> kdfSalt);
+
+    /**
+     * @brief Sets the salt for the message authentication code.
+     *
+     * If no value is set, the MAC calculation is done without any salt
+     * @param macSalt The salt value used for message authentication code.
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setMACSalt(std::vector<uint8_t> macSalt);
+
+    /**
+     * @brief Create the ECIES encryption context.
+     *
+     * Call this function after you have set the options, which differ from the default value
+     * @throws MoCOCrWException If a required parameter is not set.
+     * @param bobsKey The public key of the receiver
+     * @return An ECIES encryption context, which can be used for encryption.
+     */
+    std::unique_ptr<ECIESEncryptionCtx> buildEncryptionCtx(const AsymmetricPublicKey bobsKey);
```

Comment:
I guess since we now grab a copy of this, this no longer needs to be const?

Reply:
Done